UI: Estimate recording time left until disk is full

cg2121 opened the pull request:
https://ideas.obsproject.com/posts/366/show-remaining-record-time-based-on-disk-space-and-current-settings

![Screenshot from 2019-03-13 05-25-09](https://user-images.githubusercontent.com/19962531/54272105-fc0e3300-4550-11e9-8eca-4760ba05c4f6.png)

SuslikV commented:
I bet there is error in this code.

Also, "N/A" can be replaced for something neutral: "-".

DDRBoxman commented on UI/window-basic-stats.hpp:
```diff
@@ -29,6 +30,10 @@ class OBSBasicStats : public QWidget {
 
+	uint64_t totalSeconds = 0;
+
+	std::vector<long double> bitrates;
```
Might be worth calling reserve() on this since we have a good idea how many samples it will have.

cg2121 replied:
It is now set to reserve(15).

DDRBoxman commented on UI/window-basic-stats.cpp:
```diff
@@ -389,6 +394,38 @@ void OBSBasicStats::Update()
+
+	bitrates.push_back(kbps);
+
+	if (totalSeconds % 30 != 0)
```
Probably makes sense to do a check here if seconds is > 30 and then reset it. This enables the TIMER_INTERVAL to change without breaking this and also we don't forever increment a variable.

cg2121 replied:
I'm now using a QTimer instead.

cg2121 commented:
I refactored this a bit to use a 30 second QTimer instead of the totalSeconds variable. I also changed "N/A" to blank, and when the recording is first started, the string is set to "Calculating.."

Vainock commented:
Could you please change "Calculating.." to "Calculating..." for consistency?

cg2121 commented:
Fixed

jp9000 commented:
Great work.
